services/consensusalgo/leanhelixconsensus/key_manager.go

```diff
@@ -13,32 +13,33 @@ import (
 	lhprotocol "github.com/orbs-network/lean-helix-go/spec/types/go/protocol"
 	"github.com/orbs-network/orbs-network-go/crypto/digest"
 	"github.com/orbs-network/orbs-network-go/crypto/hash"
+	"github.com/orbs-network/orbs-network-go/crypto/kms"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/scribe/log"
 	"github.com/pkg/errors"
 )
 
 type keyManager struct {
-	privateKey primitives.EcdsaSecp256K1PrivateKey
-	logger     log.Logger
+	signer kms.Signer
+	logger log.Logger
 }
 
 // TODO Fix according to branch lh-outline, see https://tree.taiga.io/project/orbs-network/us/566
 
-func NewKeyManager(logger log.Logger, privateKey primitives.EcdsaSecp256K1PrivateKey) *keyManager {
+func NewKeyManager(logger log.Logger, signer kms.Signer) *keyManager {
 	return &keyManager{
-		logger:     logger,
-		privateKey: privateKey,
+		logger: logger,
+		signer: signer,
 	}
 }
 
 func (km *keyManager) SignConsensusMessage(blockHeight lhprimitives.BlockHeight, content []byte) lhprimitives.Signature {
-	sig, _ := digest.SignAsNode(km.privateKey, content) // TODO(v1): handle error (log) https://tree.taiga.io/project/orbs-network/us/603
+	sig, _ := km.signer.Sign(content) // TODO(v1): handle error (log) https://tree.taiga.io/project/orbs-network/us/603
```

Comment:
we need to understand the implications of failure here as we now have to deal with HTTP errors